app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsFragment.kt

```diff
@@ -242,17 +244,35 @@ class PaymentMethodsFragment : DaggerFragment(), PaymentMethodsView {
       payment_method_description.text = getPaymentMethodLabel(paymentMethod)
       payment_method_secondary.visibility = View.VISIBLE
       payment_method_description_single.visibility = View.GONE
+      payment_method_fee.visibility = View.GONE
       if (isBonusActive) hideBonus()
     } else {
       payment_method_description.visibility = View.VISIBLE
       payment_method_description.text = getPaymentMethodLabel(paymentMethod)
       payment_method_secondary.visibility = View.GONE
       payment_method_description_single.visibility = View.GONE
       if (isBonusActive) showBonus()
+      setupFee(paymentMethod.fee)
     }
     loadIcons(paymentMethod, payment_method_ic)
   }
 
+  private fun setupFee(fee: PaymentMethodFee) {
+    if (fee.active) {
+      payment_method_fee.visibility = View.VISIBLE
+      val formattedValue = CurrencyFormatUtils.create()
+          .formatCurrency(fee.amount, WalletCurrency.FIAT)
+      payment_method_fee_value.text = "$formattedValue ${fee.currency}"
+
+      payment_method_fee_value.apply {
+        this.setTextColor(ContextCompat.getColor(requireContext(), R.color.appc_pink))
+        this.typeface = Typeface.create("sans-serif-medium", Typeface.NORMAL)
```

Comment:
Why is this not done in the layout

Reply:
Because depending on enabled/disabled we need to show different typefaces and colors